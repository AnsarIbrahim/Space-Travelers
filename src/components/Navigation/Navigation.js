import React from 'react';
import { NavLink } from 'react-router-dom';
import planet from './planet.png';

export default function Navigation() {
  return (
    <nav>
      <img src={planet} alt="" />
      <h3>Space Travelers&apos; Hub</h3>
      <li className="links">
        <NavLink
          to="/"
          className="link"
        >
          <span className="my-link">Rockets</span>
        </NavLink>
      </li>
      <NavLink to="/missions">
        <span className="my-link">Missions</span>
        {' '}
      </NavLink>
      <span className="ver-line" />
      <NavLink to="/myprofile"><span className="my-link">My Profile</span></NavLink>

    </nav>
  );
}
