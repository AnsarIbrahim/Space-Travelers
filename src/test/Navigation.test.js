import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom';
import Navigation from '../components/Navigation/Navigation';

describe('Navigation Component', () => {
  it('renders correctly without unexpected changes', () => {
    const navigation = renderer
      .create(
        <Router>
          <Navigation />
        </Router>,
      )
      .toJSON();
    expect(navigation).toMatchSnapshot();
  });

  it('renders all the navigation links', () => {
    render(<MemoryRouter><Navigation /></MemoryRouter>);
    screen.debug();
    const rocketLink = screen.getByText('Rockets');
    const missionLink = screen.getByText('Missions');
    const profileLink = screen.getByText('My Profile');

    expect(rocketLink).toBeInTheDocument();
    expect(missionLink).toBeInTheDocument();
    expect(profileLink).toBeInTheDocument();
  });
});
