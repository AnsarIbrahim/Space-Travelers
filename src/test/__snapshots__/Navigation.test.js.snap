// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders correctly without unexpected changes 1`] = `
<nav>
  <img
    alt=""
    src="planet.png"
  />
  <h3>
    Space Travelers' Hub
  </h3>
  <li
    className="links"
  >
    <a
      aria-current="page"
      className="link active"
      href="/"
      onClick={[Function]}
    >
      <span
        className="my-link"
      >
        Rockets
      </span>
    </a>
  </li>
  <a
    className=""
    href="/missions"
    onClick={[Function]}
  >
    <span
      className="my-link"
    >
      Missions
    </span>
     
  </a>
  <span
    className="ver-line"
  />
  <a
    className=""
    href="/myprofile"
    onClick={[Function]}
  >
    <span
      className="my-link"
    >
      My Profile
    </span>
  </a>
</nav>
`;
